internal/pkg/plugin/gitlabcedocker/gitlabcedocker.go

```diff
@@ -1,51 +1,37 @@
 package gitlabcedocker
 
 import (
+	"strconv"
 	"strings"
 
-	"github.com/devstream-io/devstream/pkg/util/docker"
-	"github.com/devstream-io/devstream/pkg/util/docker/dockersh"
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
+	"github.com/devstream-io/devstream/pkg/util/log"
 )
 
 const (
 	gitlabImageName       = "gitlab/gitlab-ce"
 	defaultImageTag       = "rc"
 	gitlabContainerName   = "gitlab"
-	tcp                   = "tcp"
 	dockerRunShmSizeParam = "--shm-size 256m"
 )
 
-func getImageNameWithTag(opt Options) string {
-	return gitlabImageName + ":" + opt.ImageTag
-}
+// gitlabURL is the access URL of GitLab.
+var gitlabURL string
 
-func defaults(opts *Options) {
-	if opts.ImageTag == "" {
-		opts.ImageTag = defaultImageTag
+func getGitLabURL(opts *Options) string {
+	accessUrl := opts.Hostname
+	if opts.HTTPPort != 80 {
```

Comment:
Unnecessary judgment, just splice it directly

Reply:
There are too many cases of url, why not let the user fill in the url directly and use validator checks?

https://github.com/go-playground/validator/blob/master/validator_test.go#L7729-L7767

Reply:
@aFlyBird0